Parse //# sourceMappingURL= comments that contain inline source maps.

mprobst opened the pull request:
Standalone source maps can already be specified using --source_map_input. This adds support for inline source maps using data URLs with base64 encoding. Inline source maps make it easier to preserve the source map / file association across build steps and file transfers.

mprobst commented:
This is based on the changes in #1971.

mprobst commented:
/CC @ChadKillingsworth

ChadKillingsworth commented:
I was kind of waiting on #1971 to get merged - makes the diff easier to read.

Of particular note is what behavior is expected if an input sourcemap is provided AND an inline sourcemap for the same file? The gulp sourcemap plugin would never do this, but we should at least know the behavior.

mprobst commented:
@ChadKillingsworth source maps in files override any from parameters. I think that makes sense - the closer to the source code itself you are, the more "binding".

Also, source map comments later in the file override earlier ones, assuming there are multiple. Also working as intended for me - we don't support arbitrary depth mapping, so taking the last one is the closest reasonable thing we can do (at least we'll correctly map one level).

ChadKillingsworth commented:
@mprobst sounds good to me.

mprobst commented:
@ChadKillingsworth PTAL, this is now rebased on the submitted `apply_input_source_maps` branch. I've also introduced an option that allows disabling the inline source map parsing, in case somebody runs into issues with it (e.g. memory usage).
